code-challenges.test.js

```diff
@@ -35,35 +35,128 @@ const num2 = 0
+// It is important to note that if one were to only write a line of code with just .toUpperCase(), the function would return a string where all of the letters within the string are capitalized which is not our goal. We only want the first letter of each string to be capitalized.
+// Taking this into consideration, we will have to use a combination of the charAt() method and the toUpperCase() method. The charAt() will take in the 0th index as its parameter since that will correspond to only the first letter of the string which will be followed up by .toUpperCase() which will capitalize only that first letter.
+// After writing that line of code, we will have to add the rest of the string to the capitalized letter so in order to accomplish this, I will concat the rest of the string by using the .slice() built-in method. This method will take in the 1st index as its parameter, since the 1st index will always correspond to the second letter of the string and the slice method will make it so that if it is only taking one parameter, it will include the rest of the string.
+// Final step is to just return the variable capitalizedArray which should make it so that our test passes.
+
+const arrayCapitalizer = (array) => {
+  let capitalizedArray = []
+  for(let i = 0; i<array.length; i++){
+    capitalizedArray.push(array[i].charAt(0).toUpperCase() + array[i].slice(1))
+  }
+  return capitalizedArray
+}
+
+// The test passed (green) :)
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const vowelTester1 = "learn"
+// const vowelTester1 = "learn"
 // Expected output: 1
-const vowelTester2 = "academy"
+// const vowelTester2 = "academy"
 // Expected output: 0
-const vowelTester3 = "challenges"
+// const vowelTester3 = "challenges"
 // Expected output: 2
 
+// a describe method that lists the name of the function OR naming of the particular test.
+describe("vowelIndexFinder", () => {
+  const vowelTester1 = "learn"
+  const vowelTester2 = "academy"
+  const vowelTester3 = "challenges"
+  // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+  it("logs the index of the first vowel.", () => {
+
+    //an expect method, nested within the test block, calling on the hello() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+    expect(vowelIndexFinder(vowelTester1)).toEqual(1)
+    expect(vowelIndexFinder(vowelTester2)).toEqual(0)
+    expect(vowelIndexFinder(vowelTester3)).toEqual(2)
+  })
+})
+
+// Running this test has failed which is what we're looking for.
+// ReferenceError: vowelIndexFinder is not defined
 
 // b) Create the function that makes the test pass.
+
+// Create a function called vowelIndexFinder that will take in one parameter, string.
```

Comment:
You are taking some good notes on you approach. On the next assessment, try keeping your notes in your personal file. And allow the pseudocode be the steps you took to get the green pass. Definitely keep your notes because they will be great to reference later. This shows you really took time to consider your logic.